src/glmtools.jl

```diff
@@ -45,26 +45,31 @@ linkfun(::SqrtLink, μ) = sqrt(μ)
 linkinv(::SqrtLink, η) = abs2(η)
 mueta(::SqrtLink, η) = 2η
 
+canonicallink(::Bernoulli) = LogitLink()
 canonicallink(::Binomial) = LogitLink()
 canonicallink(::Gamma) = InverseLink()
 canonicallink(::Normal) = IdentityLink()
 canonicallink(::Poisson) = LogLink()
 
 # For the "odd" link functions we evaluate the linear predictor such that mu is closest to zero where the precision is higher
-function glmvar(::Binomial, link::@compat(Union{CauchitLink,InverseLink,LogitLink,ProbitLink}), μ, η)
+function glmvar(::@compat(Union{Bernoulli,Binomial}), link::@compat(Union{CauchitLink,InverseLink,LogitLink,ProbitLink}), μ, η)
     μ = linkinv(link, ifelse(η < 0, η, -η))
     μ * (1 - μ)
 end
-glmvar(::Binomial, ::Link, μ, η) = μ * (1 - μ)
+glmvar(::@compat(Union{Bernoulli,Binomial}), ::Link, μ, η) = μ * (1 - μ)
 glmvar(::Gamma, ::Link, μ, η) = abs2(μ)
 glmvar(::Normal, ::Link, μ, η) = 1
 glmvar(::Poisson, ::Link, μ, η) = μ
 
+mustart(::Bernoulli, y, wt) = 0.5y + oftype(y, 0.25)
 mustart(::Binomial, y, wt) = (wt * y + oftype(y, 0.5)) / (wt + one(y))
 mustart(::Gamma, y, wt) = y
 mustart(::Normal, y, wt) = y
 mustart(::Poisson, y, wt) = y + oftype(y, 0.1)
 
+function devresid(::Bernoulli, y, μ, wt)
+    -2wt * (y == 1 ? log(μ) : (y == 0 ? log1p(-μ) : throw(ArgumentError("$y ∉ support(Bernoulli)"))))
```

Comment:
Why not write this as an `if... elseif... else`? That would be easier to read.

Also, the error message would probably be clearer as something like `"y should be 0 or 1 (got $y)"`.